gamemodes/terrortown/gamemode/client/cl_vskin/vgui/dweaponpreview_ttt2.lua

```diff
@@ -0,0 +1,381 @@
+
+			if not IsValid(cEnt) then continue end
+
+			self.data.worldModels[name] = cEnt
+		end
+	else
+		local cEnt = ClientsideModel(wep.WorldModel, RENDERGROUP_OTHER)
+
+		if not IsValid(cEnt) then return end
+
+		cEnt:SetNoDraw(true)
+		cEnt:SetIK(false)
+
+		cEnt:SetParent(self.data.ply)
+		cEnt:AddEffects(EF_BONEMERGE)
+
+		if wep.HoldType ~= "normal" then
+			self.data.wep = cEnt
+		end
+	end
+
+
+	self.data.HoldType = wep.HoldType
+
+	if wep.HoldType == "normal" then
+		self.data.ply:SetSequence(self.data.ply:LookupSequence("idle_all_01"))
+	else
+		self.data.ply:SetSequence(self.data.ply:LookupSequence("idle_" .. wep.HoldType))
+	end
+
+	self.data.ply:SetupBones()
+end
+
+---
+-- @realm client
+function PANEL:DrawModel()
+	local ply = self.data.ply
+	local wep = self.data.wep
+
+	if not IsValid(ply) then return end
+
+	local w, h = self:GetSize()
+	local xBaseStart, yBaseStart = self:LocalToScreen(0, 0)
+
+	local xLimitStart, yLimitStart = xBaseStart, yBaseStart
+	local xLimitEnd, yLimitEnd = self:LocalToScreen(self:GetWide(), self:GetTall())
+
+	local curparent = self
+
+	-- iterate till the top is found to make sure the image is not out of bounds
+	while curparent:GetParent() do
+		curparent = curparent:GetParent()
+
+		local x1, y1 = curparent:LocalToScreen(0, 0)
+		local x2, y2 = curparent:LocalToScreen(curparent:GetWide(), curparent:GetTall())
+
+		xLimitStart = mathMax(xLimitStart, x1)
+		yLimitStart = mathMax(yLimitStart, y1)
+		xLimitEnd = mathMin(xLimitEnd, x2)
+		yLimitEnd = mathMin(yLimitEnd, y2)
+	end
+
+	-- This adds support for the "SWEP Construction Kit" that is used by many weapons such as the
+	-- boomerang or the melon mine: https://steamcommunity.com/sharedfiles/filedetails/?id=109724869
+	--
+	-- This code is taken and slightly modified from its WorldModel renderer
+	if self.data.specialModelHandeling then
+		local wepEnt = weapons.Get(self.data.wepClass)
+		local pos, ang
+
+		for name, worldModelData in pairs(self.data.worldModelData) do
+			wep = self.data.worldModels[name]
+
+			if worldModelData.bone then
+				pos, ang = wepEnt:GetBoneOrientation(wepEnt.WElements, worldModelData, self.data.ply)
+			else
+				pos, ang = wepEnt:GetBoneOrientation(wepEnt.WElements, worldModelData, self.data.ply, "ValveBiped.Bip01_R_Hand")
+			end
+
+			wep:SetPos(pos + ang:Forward() * worldModelData.pos.x + ang:Right() * worldModelData.pos.y + ang:Up() * worldModelData.pos.z)
+
+			ang:RotateAroundAxis(ang:Up(), worldModelData.angle.y)
+			ang:RotateAroundAxis(ang:Right(), worldModelData.angle.p)
+			ang:RotateAroundAxis(ang:Forward(), worldModelData.angle.r)
+
+			wep:SetAngles(ang)
+
+			local matrix = Matrix()
+			matrix:Scale(worldModelData.size)
+			wep:EnableMatrix("RenderMultiply", matrix)
+
+			if worldModelData.material == "" then
+				wep:SetMaterial("")
+			elseif wep:GetMaterial() ~= worldModelData.material then
+				wep:SetMaterial( worldModelData.material )
+			end
```

Comment:
This seems like it does just this:
```suggestion
			wep:SetMaterial( worldModelData.material )
```

Reply:
Nope, that's not the same

Reply:
But the first statement sets it to `""` if material is `""`. 
And the second one sets the material always if it is not the material.
Therefore they are both always setting worldModelData.material.

Reply:
I agree with you, I just thought it has a reason this is done that way. If issues arise, we can point it to that change